### PR TITLE
WIP: adding  LBaaS

### PR DESCRIPTION
**Pull request by chelios**

### CODE REVIEW

**Comment by chelios** [comment]
@fincham thanks for the feedback, I will revisit. 

This branch is not yet ready to be merged it is simply here so that others in the team can review the content.

File: source/lbaas.rst [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+s also known as “request switching,” “application load balancing,” “content based routing or
+switching,”
+
+A layer 7 load balancer consists of a listener that accepts requests on behalf of a number of
+back-end pools and distributes those requests based on policies that use application data to
+determine which pools should service any given request. This allows for the application
+infrastructure to be specifically tuned/optimized to serve specific types of content. For example,
+one group of back-end servers (pool) can be tuned to serve only images, another for execution of
+server-side scripting languages like PHP and ASP, and another for static content such as HTML, CSS,
+and JavaScript.
```
Comment (teolupus) [inline source/lbaas.rst]: I'd prefer a more realistic example here, such as "mydomain.nz/login or a subdomain login.mydomain.nz will be routed to a back-end pool running an identity provider and authentication system, while mydomain.nz/shop or shop.mydomain.nz will be routed to a commerce application".

File: source/lbaas.rst [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+  | operating_status          | OFFLINE                              |
+  | project_id                | a3a9af91b9e547739bfcb02cc2acded0     |
+  | protocol                  | HTTP                                 |
+  | protocol_port             | 80                                   |
+  | provisioning_status       | PENDING_CREATE                       |
+  | sni_container_refs        | []                                   |
+  | updated_at                | None                                 |
+  +---------------------------+--------------------------------------+
+
+create the first pool
```
Comment (teolupus) [inline source/lbaas.rst]: "Create" instead of create

File: source/lbaas.rst [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+  | name                | http_pool                            |
+  | operating_status    | OFFLINE                              |
+  | project_id          | a3a9af91b9e547739bfcb02cc2acded0     |
+  | protocol            | HTTP                                 |
+  | provisioning_status | PENDING_CREATE                       |
+  | session_persistence | None                                 |
+  | updated_at          | None                                 |
+  +---------------------+--------------------------------------+
+
+add the member to the pool
```
Comment (teolupus) [inline source/lbaas.rst]: Add

File: source/lbaas.rst [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+  | protocol_port       | 80                                   |
+  | provisioning_status | PENDING_CREATE                       |
+  | subnet_id           | 1c221166-3cb3-4534-915a-b75220ec1873 |
+  | updated_at          | None                                 |
+  | weight              | 1                                    |
+  | monitor_port        | None                                 |
+  | monitor_address     | None                                 |
+  +---------------------+--------------------------------------+
+
+create the second pool
```
Comment (teolupus) [inline source/lbaas.rst]: Create

File: source/lbaas.rst [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+  | name                | http_pool_2                          |
+  | operating_status    | OFFLINE                              |
+  | project_id          | a3a9af91b9e547739bfcb02cc2acded0     |
+  | protocol            | HTTP                                 |
+  | provisioning_status | PENDING_CREATE                       |
+  | session_persistence | None                                 |
+  | updated_at          | None                                 |
+  +---------------------+--------------------------------------+
+
+add the other member to the second pool
```
Comment (teolupus) [inline source/lbaas.rst]: Add

File: source/lbaas.rst [hunk 6]
```diff
@@ -0,0 +1,521 @@
...
+  | protocol_port       | 80                                   |
+  | provisioning_status | PENDING_CREATE                       |
+  | subnet_id           | 1c221166-3cb3-4534-915a-b75220ec1873 |
+  | updated_at          | None                                 |
+  | weight              | 1                                    |
+  | monitor_port        | None                                 |
+  | monitor_address     | None                                 |
+  +---------------------+--------------------------------------+
+
+create the layer 7 policy
```
Comment (teolupus) [inline source/lbaas.rst]: Create

File: source/lbaas.rst [hunk 7]
```diff
@@ -0,0 +1,521 @@
...
+  | redirect_pool_id    | af13eb62-d4a1-44e5-8a9d-d7df0595b8bb |
+  | redirect_url        | None                                 |
+  | action              | REDIRECT_TO_POOL                     |
+  | position            | 1                                    |
+  | id                  | 7b191c4f-cc22-4896-8b16-0c703d8b5220 |
+  | operating_status    | OFFLINE                              |
+  | name                | policy1                              |
+  +---------------------+--------------------------------------+
+
+create a rule for the policy
```
Comment (teolupus) [inline source/lbaas.rst]: Create

**Pull request merged**